datafusion/core/src/config.rs

```diff
@@ -360,6 +360,11 @@ impl ConfigOptions {
         self.set(key, ScalarValue::UInt64(Some(value)))
     }
 
+    /// set a `String` configuration option
+    pub fn set_string(&mut self, key: &str, value: String) {
```

Comment:
I recommend using an `impl Into<String>` so callers an pass a `&str` as well as a `String`, depending on what they have


```suggestion
    pub fn set_string(&mut self, key: &str, value: impl Into<String>) {
```

Reply:
@alamb thank you, this looks great, didn't know this before. I'll attach another commit soon